### PR TITLE
Azure support

### PR DESCRIPTION
**Pull request by jlowin**

### CODE REVIEW

File: cookbook/azure/README.md [hunk 1]
```diff
@@ -6,19 +6,22 @@ It is possible to use Azure OpenAI with _some_ of `marvin`'s functionality via:
 !!! Note
     Azure OpenAI often lags behind the latest version of OpenAI in terms of functionality, therefore some features may not work with Azure OpenAI. If you encounter problems, please check that the underlying functionality is supported by Azure OpenAI before reporting an issue.
 
-## Settings
-After setting up your Azure OpenAI account and deployment, it is recommended you save these settings in your `~/.marvin/.env` file.
+## Configuring with environment variables
+After setting up your Azure OpenAI account and deployment, set these environment variables in your environment or your `~/.marvin/.env` file:
```
Comment (zzstoatzz) [inline cookbook/azure/README.md:10]:
```suggestion
After setting up your Azure OpenAI account and deployment, set these environment variables in your environment, your `~/.marvin/.env` or `.env` file:
```

File: src/marvin/utilities/openai.py [hunk 2]
```diff
@@ -22,33 +28,90 @@ def get_openai_client() -> AsyncClient:
...
-    organization: Optional[str] = settings.openai.organization
+    kwargs = {}
+
+    # --- Openai
+    if marvin.settings.provider == "openai":
+        client_class = AsyncClient if is_async else Client
+
+        api_key = marvin.settings.openai.api_key
+        if isinstance(api_key, pydantic.SecretStr):
+            api_key = api_key.get_secret_value()
```
Comment (zzstoatzz) [inline src/marvin/utilities/openai.py:40]: nit: is this check needed? im pretty sure no matter how you set it, [it should be a `SecretStr`](https://github.com/PrefectHQ/marvin/blob/main/src/marvin/settings.py#L20-L32) by the time we look here

**Review by zzstoatzz — APPROVED** [review]

File: src/marvin/utilities/openai.py (hunk 2 above)
Comment (jlowin) [reply src/marvin/utilities/openai.py:40]: Empirically yes it is but now that you mention it I don't know why... but I can get this to crash without that. Let me look into it, something must be weird

Comment (jlowin) [reply src/marvin/utilities/openai.py]: https://github.com/PrefectHQ/marvin/pull/772/commits/0d993d6f178af8ee1b1dd57a96cb13cea2a44a4b

**Pull request merged**